libvast/vast/concept/parseable/to.hpp

```diff
@@ -19,18 +19,17 @@
 namespace vast {
 
 template <class To, class Iterator>
-auto to(Iterator& f, const Iterator& l)
-  -> std::enable_if_t<is_parseable_v<Iterator, To>, caf::expected<To>> {
+requires(parseable<Iterator, To>) auto to(Iterator& f, const Iterator& l)
+  -> caf::expected<To> {
   caf::expected<To> t{To{}};
   if (!parse(f, l, *t))
     return caf::make_error(ec::parse_error);
   return t;
 }
 
 template <class To, class Range>
```

Comment:
`Range` would also make for a great concept, with a specialization for character ranges for this function.

Reply:
Agreed. I'm starting to see more places where good concepts could help as I'm perusing more of the codebase with my C++20 lens in mind.